perftracker/templates/comparison_id.html

```diff
@@ -161,16 +164,106 @@ <h4>Job: {{ job }}</h4>
         var s = sect_data[s_id];
         if (group_data[s.g_id].collapsed)
             return; // we have to avoid rendering collapsed charts - the rendering is wrong
-        if (!s.chart_rendered && s.chart_type !== NOCHART && pt_isInViewport($("#chart_" + s_id)[0])) {
+        if (!s.chart_rendered && s.chart_type !== NOCHART && pt_isInViewport($(`#chart_${s_id}`)[0])) {
             show_chart(s_id);
             // console.log(s_id);
         }
-        if (!s.table_rendered && s.table_type !== NOTABLE && !s.table_hidden && pt_isInViewport($("#results_" + s_id)[0])) {
+        if (!s.table_rendered && s.table_type !== NOTABLE && !s.table_hidden && pt_isInViewport($(`#results_${s_id}`)[0])) {
             show_table(s_id);
         }
     });
 }
 
+function pt_charts_analyze() {
+    if (!DEBUG_MODE) {
+        alert('Not available in production mode!');
+        return;
+    }
+    $('#analyze_button').hide('fast');
+
+    var payload = {};
+    Object.keys(sect_data).forEach(s_id => {
+        payload[s_id] = sect_data[s_id].series[0].data;
+    });
+
+    $.ajax({
+        url: api_root + "/analyze",
+        contentType: "application/json",
+        data: JSON.stringify(payload),
+        type: 'POST',
+        timeout: 10000,
+        success: function (response) {
+            Object.keys(sect_data).forEach(s_id => {
+                sect_data[s_id].info = response[s_id];
+                $(`#results_${s_id}`).append(`<p>${JSON.stringify(response[s_id])}</p>`).hide().show('fast');
+            });
+        },
+        error: function (data, status, error) {
+            console.log(error);
+        }
+    });
+}
+
+function pt_charts_label_data(s_id) {
+
+    if (sect_data[s_id].series.length > 1) {
+        console.log('Method not allowed');
+        return;
+    }
+
+    const survey = $(`#survey_${s_id}`);
+
+    if (survey.children().length !== 0) {
+        survey.is(':visible') ? survey.hide('fast') : survey.show('fast');
+        return;
+    }
+
+    Object.entries(graph_properties).forEach(([property, options]) => {
+        const selector_id = `${s_id}_${property}_selector`;
+        survey.append(`<p id="${selector_id}" class="pt_data_train"><b>${property}</b></p>`).hide().show('fast');
+        var labels = '';
+        options.forEach(option => {
+            labels += `<input type="radio" id="${selector_id}_${option}" name="${property}" value="${option}">` +
+                `<label for="${selector_id}_${option}">${option}</label>`;
+        });
+        $(`#${selector_id}`).append(labels).hide().show('fast');
+    });
+
+    survey.append(`<button onclick="pt_save_graph_properties(${s_id})"><b>Save</b></button>`);
+}
+
+function pt_save_graph_properties(s_id) {
+    labeled_data[s_id] = {points: sect_data[s_id].series[0].data};
```

Comment:
important: is it usually important for analysis whether higher is better or lower is better.
We have this information in DB, but not in browser and do not pass here